pkg/volume/csi/csi_metrics_test.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"io"
+	"testing"
+
+	csipbv1 "github.com/container-storage-interface/spec/lib/go/csi"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/volume/csi/fake"
+)
+
+func TestGetMetrics(t *testing.T) {
+
+	tests := []struct {
+		name          string
+		volumeID      string
+		targetPath    string
+		expectSuccess bool
+	}{
+		{
+			name:          "with valid name and volume id",
+			expectSuccess: true,
+			volumeID:      "foobar",
+			targetPath:    "/mnt/foo",
+		},
+	}
+
+	for _, tc := range tests {
+		metricsGetter := &metricsCsi{volumeID: tc.volumeID, targetPath: tc.targetPath}
+		metricsGetter.csiClient = &csiDriverClient{
+			driverName: "com.google.gcepd",
+			nodeV1ClientCreator: func(addr csiAddr) (csipbv1.NodeClient, io.Closer, error) {
+				nodeClient := fake.NewNodeClientWithVolumeStats(true /* VolumeStatsCapable */)
+				fakeCloser := fake.NewCloser(t)
+				nodeClient.SetNodeVolumeStatsResp(getRawVolumeInfo())
+				return nodeClient, fakeCloser, nil
+			},
+		}
+		metrics, err := metricsGetter.GetMetrics()
+		if err != nil {
+			t.Fatalf("for %s: unexpected error : %v", tc.name, err)
+		}
+		if metrics == nil {
+			t.Fatalf("unexpected nil metrics")
+		}
+		expectedMetrics := getRawVolumeInfo()
+		for i, usage := range expectedMetrics.Usage {
+			if usage.Unit == csipbv1.VolumeUsage_BYTES {
+				availableBytes := resource.NewQuantity(expectedMetrics.Usage[i].Available, resource.BinarySI)
+				totalBytes := resource.NewQuantity(expectedMetrics.Usage[i].Total, resource.BinarySI)
```

Comment:
can we replace `expectedMetrics.Usage[i].Available` with `usage.Available` ?

Reply:
Done.